generator/sidekick/config.go

```diff
@@ -14,6 +14,13 @@
 
 package main
 
+import (
+	"fmt"
+	"os"
+
+	toml "github.com/pelletier/go-toml/v2"
+)
+
 type Config struct {
 	General GeneralConfig `toml:"general"`
```

Comment:
What's the difference between Config and General Config?

Reply:
I am terrible at naming.

There are things that affect (or or at least most) parsers and codecs.  They go into `GeneralConfig`.  There are things that are only codec-specific (they go into the `codec` section).

Happy to rename all these things if you have better suggestions.

Reply:
Also added some comments for "future us".